R/functions_netCDF.R

```diff
@@ -1,333 +1,420 @@
+    }
   }
 
-
-  #--- add attributes
-  # add dimension attributes
-  ncdf4::ncatt_put(nc, "lon", "axis", "X")
-  ncdf4::ncatt_put(nc, "lon", "bounds", "lon_bnds")
-  ncdf4::ncatt_put(nc, "lat", "axis", "Y")
-  ncdf4::ncatt_put(nc, "lat", "bounds", "lat_bnds")
-
-  if (has_time_central) {
-    ncdf4::ncatt_put(nc, "time", "axis", "T")
-    ncdf4::ncatt_put(nc, "time", "bounds", "time_bnds")
-    ncdf4::ncatt_put(nc, "time", "calendar", "gregorian")
+  # ---------------------------------------------------------------------
+  # Setup  --------------------------------------------------------------
+  # ---------------------------------------------------------------------
+  if (nl == 1 && is.null(dim(data))) {
+    data <- matrix(data, ncol = 1, dimnames = list(NULL, names(data)))
   }
 
-  # add global attributes
-  ncdf4::ncatt_put(nc, varid = 0, attname = "Conventions", attval = "CF-1.4")
-  ncdf4::ncatt_put(nc, varid = 0, attname = "created_by",
-    attval = paste0(R.version[["version.string"]], ", R packages ",
-      if (requireNamespace("rSFSW2")) {
-        paste0("rSFSW2 v", utils::packageVersion("rSFSW2"), " and ")
-      },
-      "ncdf4 v", utils::packageVersion("ncdf4"),
-      ", and ", system2("nc-config", "--version", stdout = TRUE, stderr = TRUE))
-    )
-  ncdf4::ncatt_put(nc, varid = 0, attname = "creation_date",
-    attval = format(Sys.time(), "%Y-%m-%d %H:%M:%S"))
-
-  if (!missing(global_attributes)) {
-    ns_att_glob <- names(global_attributes)
+  ncdf4_datatype <- raster:::.getNetCDFDType(get_raster_datatype(data))
 
-    for (natt in ns_att_glob) {
-      ncdf4::ncatt_put(nc, varid = 0, attname = natt,
-        attval = global_attributes[[natt]])
+  NAflag <- switch(ncdf4_datatype,
+                   char = NULL, byte = NULL, short = -128L, integer = -2147483647L,
+                   float = -3.4e+38, double = -1.7e+308)
+
+  # location and spatial info  -------------------------------------------------------------
+  if(!missing(locations)) {
+    if (inherits(locations, 'sf')) {
+      loc <- sf::st_geometry(locations)
+      st_crs(loc) <- crs 
+    } else { # assume spatial or turn to spatial
+      loc <- sp::SpatialPoints(locations)
+      sp::proj4string(loc) <- sp::CRS(crs)
+      if(isGridded) sp::gridded(loc) = TRUE
     }
   }
 
-  if (!has_time_central) {
-    ncdf4::ncatt_put(nc, varid = 0, attname = "time_label",
-      attval = "None")
-    ncdf4::ncatt_put(nc, varid = 0, attname = "time_title",
-      attval = "No temporal dimensions ... fixed field")
+  if(!isGridded) {
+    if(inherits(loc, 'Spatial')) {
+      nloc <- dim(loc@coords)[1]
+    } 
+    if(inherits(loc, 'sf')) {
+      nloc <- dim(locations)[1]
+    } 
   }
 
-  # add coordinate system attributes
-  prj <- raster::crs(x)
-  if (!is.na(prj)) {
-    ncdf4::ncatt_put(nc, varid = "crs", attname = "proj4",
-      attval = as.character(prj))
+  # Note: xvals should be organized from west to east, yvals from north to south
```

Comment:
Why are the `yvals [sorted] from north to south`, i.e., decreasingly? This seems counter-intuitive to me. I don't see anything in the CF standard that this is required: http://cfconventions.org/cf-conventions/cf-conventions.html#latitude-coordinate

Reply:
You are correct. This is just legacy code.